libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,443 @@
+  @ViewChild('thumb', { static: true })
+  _thumb: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider fill */
+  @ViewChild('sliderFill', { static: true })
+  _sliderFill: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider background */
+  @ViewChild('sliderBackground', { static: true })
+  _sliderBackground: ElementRef<HTMLDivElement>;
+
+  /** @internal Holds the slider input field */
+  @ViewChild(DtInput, { static: true })
+  _sliderInput: DtInput;
+
+  /** Observer that completes on ngOnDestroy */
+  private _destroy$ = new Subject<void>();
+
+  constructor(
+    private _changeDetectionRef: ChangeDetectorRef,
+    private _zone: NgZone,
+    private _platform: Platform,
+  ) {}
+
+  ngOnInit(): void {
+    const initValue = roundToSnap(this.value, this.step, this._min, this._max);
+    if (this._value$.value !== initValue) {
+      this._updateValue(initValue);
+    }
+  }
+
+  ngAfterViewInit(): void {
+    if (this._platform.isBrowser && 'ResizeObserver' in window) {
+      this._observer = new window.ResizeObserver(_ => {
+        this._resizeObserver$.next();
+      });
+
+      this._observer.observe(this._trackWrapper.nativeElement);
+    }
+
+    this._clientRect$ = merge(
+      this._resizeObserver$.pipe(debounceTime(50)),
+      of(null), // at least one initial trigger is needed
+    ).pipe(
+      map(() => this._trackWrapper.nativeElement.getBoundingClientRect()),
+      takeUntil(this._destroy$),
+    );
+
+    this._zone.runOutsideAngular(() => {
+      this._captureEvents();
+    });
+  }
+
+  ngOnDestroy(): void {
+    this._destroy$.next();
+    this._destroy$.complete();
+    if (this._platform.isBrowser && this._observer) {
+      this._observer.disconnect();
+    }
+  }
+
+  /** @internal Updates The slider based on the new value */
+  private _updateSlider(value: any): void {
+    this._trackWrapper.nativeElement.setAttribute(
+      'aria-valuenow',
+      value.toString(),
+    );
+    this._updateInput(value);
+    this._updateValue(value, false);
+    this._updateSliderPosition(value, this._min, this._max);
+  }
+
+  /** Updates the input field with new value. */
+  private _updateInput(value: number): void {
+    this._sliderInput.value = value.toString();
+  }
+
+  /** Update the slider thumb position based on value, min and max. */
+  private _updateSliderPosition(value: number, min: number, max: number): void {
+    const position: number = getSliderPositionBasedOnValue({ value, min, max });
+    this._thumb.nativeElement.style.transform = `translateX(-${100 -
+      position * 100}%)`;
+    this._sliderFill.nativeElement.style.transform = `scale3d(${position}, 1, 1)`;
+    this._sliderBackground.nativeElement.style.transform = `scale3d(${1 -
+      position}, 1, 1)`;
+  }
+
+  /**
+   * Capture user events, and implement necessary streams to update view.
+   * TODO: refactor this to have cleaner streams, and possibly refactor into smaller functions.
```

Comment:
I think this is already as clean as it gets :) Good job!

Reply:
All right, thanks, I'll remove the comment then.